code/game/objects/items/weapons/chemical_flamethrower/fire_effect.dm

```diff
@@ -0,0 +1,91 @@
+GLOBAL_LIST_EMPTY(flame_effects)
+#define MAX_FIRE_EXIST_TIME 10 MINUTES // That's a lot of fuel, but you are not gonna make it last for longer
+
+/obj/effect/fire
+	name = "\improper Fire"
+	desc = "You don't think you should touch this."
+	icon = 'icons/effects/chemical_fire.dmi'
+	icon_state = "fire1"
+
+	/// How hot is our fire?
+	var/temperature
+	/// How long will our fire last
+	var/duration = 10 SECONDS
+	/// How many firestacks does the fire give to mobs
+	var/application_stacks = 1
+
+/obj/effect/fire/Initialize(mapload, reagent_temperature, reagent_duration, fire_applications)
+	. = ..()
+
+	if(!reagent_duration || !reagent_temperature) // There is no reason for this thing to exist
+		qdel(src)
+		return
+
+	duration = reagent_duration
+	temperature = reagent_temperature
+	application_stacks = max(application_stacks, fire_applications)
+
+	for(var/obj/effect/fire/flame as anything in GLOB.flame_effects)
+		if(flame == src)
+			continue
+		if(get_dist(src, flame) < 1) // It's on the same turf
+			merge_flames(flame)
+
+	for(var/atom/movable/thing_to_burn in get_turf(src))
```

Comment:
Should this maybe happen on process, instead of just initialize?

Reply:
You're right, done!